ledger/distribution/src/main.rs

```diff
@@ -203,6 +183,45 @@ impl BlockHandler for S3BlockWriter {
                 .expect("failed to serialize ArchiveBlocks"),
         );
     }
+
+    fn block_exists(&self, block_index: BlockIndex) -> bool {
+        let runtime = Handle::current();
+        let result = retry(
+            delay::Exponential::from_millis_with_base_factor(10).map(delay::jitter),
+            || {
+                let dest = self.path.join(block_num_to_s3block_path(block_index));
+
+                let dir = dest
+                    .parent()
+                    .expect("failed getting parent")
+                    .to_string_lossy();
+                let filename = dest
+                    .file_name()
+                    .expect("Failed getting the file name")
+                    .to_string_lossy();
+                let req = HeadObjectRequest {
+                    bucket: dir.into(),
+                    key: filename.into(),
+                    ..Default::default()
+                };
+                log::info!(self.logger, "Checking for existence of block {block_index}");
+
+                let result = runtime.block_on(self.s3_client.head_object(req));
+                match result {
+                    Ok(_) => OperationResult::Ok(true),
+                    Err(RusotoError::Service(HeadObjectError::NoSuchKey(_))) => {
+                        OperationResult::Ok(false)
+                    }
+                    Err(e) => {
+                        log::warn!(self.logger, "Failed to talk to S3 {e:?}, retrying...");
+                        OperationResult::Retry(e)
+                    }
```

Comment:
Unfortunately rusoto won't make this call without AWS credentials
So I have not been able to test this out locally and I'm not sure the CI/CD will fully exercise.
This may need to be tested out in a dev environment somehow